libs/collectives/include/hpx/collectives/all_reduce.hpp

```diff
@@ -444,7 +444,7 @@ namespace hpx { namespace lcos
 
         return all_reduce(
             hpx::find_from_basename(std::move(name), root_site),
-            std::move(local_result), std::forward<F>(op), this_site);
+            std::forward(local_result), std::forward<F>(op), this_site);
```

Comment:
@hkaiser could you please check that the move here and in `all_to_all.hpp` weren't intentional?

Reply:
The `forward` looks like the right thing to use. Thanks for fixing this!